### ISSUE TITLE
Ratings

### ISSUE THREAD

**Issue by purva98** [hovancik/stretchly#200]
Can we implement a feature which rates a person's performance based on the number of breaks he/she has skipped ?

**Comment by purva98** [comment]
I and @riddhi99 were thinking to work on it , we wanted to ask if giving ratings is a good idea?

**Comment by hovancik** [comment]
Hi, this is on my TODO list, but it's more complicated feature and I am still thinking about how to do it. Right now, this is how I imagine it: 

 - stretchly has events: break has started, break has finished, user has entered idle time, idle time ended, system went to sleep, wake up of the system, etc... 
- every of this event is saved into local DB file (eg. `Event`)
- when we are saving event, and we know it's the end of the event (eg end of break), we save new line in DB, let's say `Activity` (regural break, idle time, skipped break, extra break...)
- from `Activities`, we can then have some stats, graphs for skipped break, idle time etc..

Does this makes sense? Non of the above is in stretchly, so we would need to find some DB and write logic to handle all those events and possibilities.

Also: there is a very old branch that I've never finished, but you might be interested in finishing it: 
https://github.com/hovancik/stretchly/tree/feature/dangerous-breaks 

Basically what is does is coloring break windows (like in game, when you are injured) once you skip the set amount of breaks.  I had some problems with CSS and Retina displays. If you are skilled in CSS, you could update the branch and finish it ;] That might be fun little feature for stretchly ;]  

Let me know what you think.

**Comment by purva98** [comment]
Thanks for the detailed explanation. We have earlier worked with DB so we can try to solve that problem. Working with css is always fun, thanks for sharing the link :)

**Comment by tsah20** [comment]
@hovancik   Can we use [JSON DB]((https://www.npmjs.com/package/node-json-db) for local storage.

**Comment by tsah20** [comment]
@hovancik Can we use [Json DB](https://www.npmjs.com/package/node-json-db) for local storage.

**Comment by hovancik** [comment]
I would rather go with something like SQLite. I'm afraid that read/write from JSI+ON file would be slow once we have a lot of data. And we will have a lot of data for sure.

**Comment by apoorvaagupta** [comment]
Hi @hovancik ,
I have worked with Electron and sqlite before! Can I give this issue a try as no one is assigned to it?!

**Comment by apoorvaagupta** [comment]
Thanks @hovancik !  Can you suggest any other issue as . I would love to contribute to this project! Also, I and my team mate bhavya are aspiring for RGSoC!

**Comment by Sveijk** [comment]
@hovancik this issue references to another issue #86. We're adding a new issue for history of breaks to start with.

**Comment by hovancik** [comment]
closing in favor of #266
